students/LouReis/Lesson04/mailroom2.py

```diff
@@ -22,113 +22,123 @@
+#         for item in amounts:
+#             total = total + item
+#         print ('{:25} ${:>15,.2f} {:>15} ${:>15,.2f}'.format(name, total, count, total/count))
+
+# Tested & modified the below code to develop the complex sorted output.
+# for key,value in sorted(donations.items(),key=lambda i:sum(i[1]),reverse=True):
+#     print (key,value)
+#     print (donations[key])
+#     total = 0
+#     total = sum(donations[key])
+#     print(total)
+
+# The following function takes the 'donations' dictionary and creates a sorted report.
+# Using a complex 'sorted' for loop.
+def calculate_sort(donations):
     print('{:25} | {:^13} | {:^13} |   {:>13}'.format('Donor Name', 'Total Given', 'Num Gifts', 'Average Gift'))
     print('---------------------------------------------------------------------------')
     print()
-    while len(donor_amount) != 0:
-        max = 0
-        max_index = 0
-        for z, item_c in enumerate(donor_amount):
-            if item_c > max:
-                max = item_c
-                max_index = z
-        print('{:25} ${:>15,.2f} {:>15} ${:>15,.2f}'.format(temp_donors[max_index], donor_amount[max_index], donor_count[max_index], donor_average[max_index]))
-        del temp_donors[max_index]
-        del donor_amount[max_index]
-        del donor_count[max_index]
-        del donor_average[max_index]
-
-# The following function takes the donation list and calculates & stores totals & averages
-def total(list, donors):
-    donor_amount = []
-    donor_count = []
-    donor_average = []
-    for x, item_a in enumerate(donors):
+    for key,value in sorted(donations.items(),key=lambda i:sum(i[1]),reverse=True):
+        total = 0
+        total = sum(donations[key])
         count = 0
-        amount = 0
-        for y, item_b in enumerate(list):
-            if item_b == donors[x]:
-                name = item_a
-                count = count + 1
-                amount = amount + list[y+1]
-        # The below print statement was for testing unsorted output.
-        # print('{:25} ${:>15} {:>15} ${:>15}'.format(name, amount, count, amount/count))
-        donor_amount.append(amount)
-        donor_count.append(count)
-        donor_average.append(amount/count)
-    print()
-    sort_max(donors, donor_amount, donor_count, donor_average)
+        count = len(donations[key])
+        print ('{:25} ${:>15,.2f} {:>15} ${:>15,.2f}'.format(key, total, count, total/count))
 
 # Below is the main menu function that continues prompting until quit.
 def main_menu(main_prompt,menu_options_dict):
+    import sys
     while True:
         response = input(main_prompt)
         if menu_options_dict[response]() == "Quit":
-            break
+            sys.exit()
 
 # Below are the 4 menu options that are declared in the dict.
-def option_one():
+def donation_report():
     print('\nYou Chose Option 1\n\n')
     print('DONATION SUMMARY REPORT\n\n')
-    total(donations,donor_list)
+    calculate_sort(donations)
```

Comment:
at this point you're not gaining anything by having a separate function to print your report; even so, the function is called `calculate_sort` however it is doing more than that.

you can move your report print code in this function (`donation_report`)